src/main/kotlin/net/irisshaders/lilybot/extensions/util/RoleMenu.kt

```diff
@@ -4,206 +4,523 @@ import com.kotlindiscord.kord.extensions.DISCORD_BLACK
+				name = "create"
+				description = "Create a new role menu in this channel. A channel can have any number of role menus."
+
+				check {
+					anyGuild()
+					hasPermission(Permission.ManageRoles)
+					configPresent()
+				}
+
+				var menuMessage: Message?
+				action {
+					val kord = this@ephemeralSlashCommand.kord
+
+					if (!botHasRolePermissions(kord) || !botCanAssignRole(kord, arguments.initialRole)) {
+						return@action
+					}
+
+					menuMessage = channel.createMessage {
+						if (arguments.embed) {
+							embed {
+								description = arguments.content
+								color = arguments.color
+							}
+						} else {
+							content = arguments.content
+						}
+					}
+
+					// While we don't normally edit in components, in this case we need the message ID.
+					menuMessage!!.edit {
+						val components = components {
+							ephemeralButton {
+								label = "Select roles"
+								style = ButtonStyle.Primary
+
+								this.id = "role-menu${menuMessage!!.id}"
+
+								action { }
+							}
+						}
+
+						components.removeAll()
+					}
+
+					DatabaseHelper.setRoleMenu(
+						menuMessage!!.id,
+						channel.id,
+						guild!!.id,
+						mutableListOf(arguments.initialRole.id)
+					)
+
+					val config = DatabaseHelper.getConfig(guild!!.id)
+					val actionLog = guild!!.getChannel(config!!.modActionLog) as GuildMessageChannelBehavior
+
+					actionLog.createMessage {
+						embed {
+							title = "Role Menu Created"
+							description = "A role menu for the ${arguments.initialRole.mention} role was created in " +
+									"${channel.mention}."
+
+							field {
+								name = "Content:"
+								value = "```${arguments.content}```"
+								inline = false
+							}
+							field {
+								name = "Color:"
+								value = arguments.color.toString()
+								inline = true
+							}
+							field {
+								name = "Embed:"
+								value = arguments.embed.toString()
+								inline = true
+							}
+						}
+						components {
+							linkButton {
+								label = "Jump to role menu"
+								url = menuMessage!!.getJumpUrl()
+							}
+						}
+					}
+
+					respond {
+						content = "Role menu created. You can add more roles using the `/role-menu add` command."
+					}
+				}
 			}
 
-			action {
-				val descriptionAppendix = "\n\nUse the button below to add/remove the ${arguments.role.mention} role."
+			ephemeralSubCommand(::RoleMenuAddArgs) {
+				name = "add"
+				description = "Add a role to the existing role menu in this channel."
 
-				val targetChannel: GuildMessageChannelBehavior =
-					if (arguments.channel != null) {
-						val targetChannelId = arguments.channel!!.id
-						guild?.getChannel(targetChannelId) as GuildMessageChannelBehavior
-					} else {
-						this.channel as GuildMessageChannelBehavior
+				check {
+					anyGuild()
+					hasPermission(Permission.ManageMessages)
+					configPresent()
+				}
+
+				action {
+					val kord = this@ephemeralSlashCommand.kord
```

Comment:
I feel like looking into a central `kord` variable would be useful

Reply:
That's for another PR to handle